mlrun/api/main.py

```diff
@@ -210,6 +220,70 @@ def _start_periodic_runs_monitoring():
         )
 
 
+def _start_periodic_pulling_of_clusterization_spec():
+    interval = int(config.httpdb.clusterization.worker.sync_with_chief.interval)
+    if interval > 0:
+        logger.info("Starting periodic chief state pulling", interval=interval)
+        run_function_periodically(
+            interval,
+            _synchronize_with_chief_clusterization_spec.__name__,
+            False,
+            _synchronize_with_chief_clusterization_spec,
+        )
+
+
+async def _synchronize_with_chief_clusterization_spec():
+    try:
+        chief_client = mlrun.api.utils.clients.chief.Client()
+        clusterization_spec = chief_client.get_clusterization_spec(
+            proxy=False, raise_on_failure=True
+        )
+    except Exception as exc:
+        logger.debug("Were unable receive clusterization spec", exc=str(exc))
+    else:
+        # TODO: Do we want to pull chief state all the time?
+        if config.httpdb.state != mlrun.api.schemas.APIStates.online:
+            chief_state = clusterization_spec.chief_api_state
+            if _is_chief_reached_online_state(clusterization_spec):
+                await move_api_to_online()
+
+                mlrun.mlconf.httpdb.state = chief_state
+                logger.info("Worker state reached online")
+                # at the moment we use this function to synchronize the worker state with the chief state,
+                # once we will use this for consistent periodic pulling we will have no need to cancel the
+                # periodic function when reaching online state
+                cancel_periodic_function(
+                    _synchronize_with_chief_clusterization_spec.__name__
+                )
+            else:
+                # we want the worker to be aligned with chief state
+                config.httpdb.state = chief_state
+
+                if config.httpdb.state == mlrun.api.schemas.APIStates.offline:
+                    logger.info(
+                        "Chief state is offline, canceling worker periodic chief clusterization spec pulling. "
+                        "Restart is required"
+                    )
+                    cancel_periodic_function(
+                        _synchronize_with_chief_clusterization_spec.__name__
+                    )
```

Comment:
The code here is very complex and I'm not sure why
1. Remove todo, maybe change to `# assumption: we can't get out of a terminal api state, so no need to continue pulling when we reached one`
2. `mlrun.api.schemas.APIStates` should have a `terminal_states` method and and then logic should be something like: 
* align my state to chief state
* if we reached terminal state -> cancel periodic
3. Remove that "restart is required", offline might be a desired state

Reply:
The main Idea was that maybe in the future we will want to use that pulling mechanism for other stuff rather than chief state. That is why I didn't call any of the triggering functions related to chief state but rather clusterization spec and then inside of it we will implement the synchronization logic that we want to apply from the chief.

Reply:
Changed part of what you mentioned, there are states I want to set only after I run a function, e.g run `move_api_to_online` and after running the function to change the API state.